modules/snooker/src/main/ts/ephox/snooker/resize/TableSize.ts

```diff
@@ -6,36 +6,40 @@ import * as Sizes from './Sizes';
 import { Warehouse } from '../model/Warehouse';
 import { BarPositions, ColInfo } from './BarPositions';
 
+export type WidthType = 'fixed' | 'relative';
+
 export interface TableSize {
   width: () => number;
   pixelWidth: () => number;
   getWidths: (warehouse: Warehouse, direction: BarPositions<ColInfo>, tableSize: TableSize) => number[];
+  getNewWidths: (widths: number[], deltas: number[]) => number[];
   getCellDelta: (delta: number) => number;
   singleColumnWidth: (w: number, delta: number) => number[];
   minCellWidth: () => number;
   setElementWidth: (cell: Element, amount: number) => void;
   setTableWidth: (table: Element, newWidths: number[], delta: number) => void;
+  widthType: WidthType;
```

Comment:
This really should not exist for branching purposes. Any branching specific code should be here instead and called.

Reply:
Also worth noting, if adding new sizing modes, then it means you need to add the branching everywhere you'd done it. So makes adding new feature or future maintenance much harder doing what you've currently done. As an example, these changes wouldn't work with my changes for the new "none" sizing because of this.